src/main/lpfConverter.ts

```diff
@@ -100,110 +100,110 @@ async function injectManifestToZip(audiobookPath: string, manifest: Buffer) {
-        debug(`not an audiobook W3C publication manifest. conformsTo=${w3cManifest.conformsTo}`);
+    {
+        const conformsTo = pop("conformsTo");
+        if (conformsTo !== "https://www.w3/org/TR/audiobooks/") {
+            debug(`not an audiobook W3C publication manifest. conformsTo=${w3cManifest.conformsTo}`);
 
-        publication.Metadata.RDFType = "https://schema.org/CreativeWork";
-    } else {
-        publication.Metadata.RDFType = "https://schema.org/Audiobook";
+            publication.Metadata.RDFType = "https://schema.org/CreativeWork";
+        } else {
+            publication.Metadata.RDFType = "https://schema.org/Audiobook";
+        }
     }
-
-    publication.Metadata.Identifier = `${w3cManifest.id || ""}` || `${w3cManifest.url || ""}` || uuidV4();
-
     {
-        // Every metadata property found in the W3C manifest is copied as-is into the Readium manifest,
-        // with the following exceptions:
-
-        const metadataPropertyExceptionInW3cManifest = [
-            "@context",
-            "conformsTo",
-            "name",
-            "duration",
-            "inLanguage",
-            "datePublished",
-            "dateModified",
-            "resources",
-            "readingOrder",
-        ];
-
-        const additionalKey = Object.keys(w3cManifest).filter(
-            (key) =>
-                !metadataPropertyExceptionInW3cManifest.includes(key),
-        );
-        if (additionalKey.length) {
-            publication.Metadata.AdditionalJSON = {};
-            additionalKey.forEach(
-                (key) =>
-                    publication.Metadata.AdditionalJSON[key] = w3cManifest[key],
-            );
-        }
+
+        publication.Metadata.Identifier = `${pop("id") || ""}` || `${pop("url") || ""}` || uuidV4();
     }
     {
-        const name = w3cManifest.name;
+
+        interface IW3cName {
+            language?: string;
+            value?: string;
+        }
+
+        const name = pop("name");
         if (typeof name === "string") {
             publication.Metadata.Title = `${w3cManifest.name}`;
 
         } else if (typeof name === "object") {
-            const nameArray = Array.isArray(name) ? name : [name];
-            const titleObj = nameArray.reduce(
-                (pv: any, cv: any) =>
-                    cv.value && cv.language ? pv[`${cv.language}`] = `${cv.value}` : pv,
-                {});
-            if (ObjectKeys(titleObj).length) {
-                publication.Metadata.Title = titleObj;
+            const nameObjArray = (Array.isArray(name) ? name : [name]) as IW3cName[];
+            const nameMap = nameObjArray.reduce(
+                (pv, cv: IW3cName) =>
+                    cv.value && cv.language ? pv.set(`${cv.language}`, `${cv.value}`) : pv,
+                new Map<string, string>());
+            if (nameMap.size) {
+                publication.Metadata.Title = Object.fromEntries(nameMap);
```

Comment:
Suggestion: get rid of `Object.fromEntries()` so that Thorium's output ES6 bundle does include TypeScript's ES2018/19 polyfill libs, and simply replace `new Map<string, string>()` with `{}` as the `array.reduce()` initial object, and of course instead of `pv.set(`${cv.language}`, `${cv.value}`)` just return `{ ...pv, [cv.language]: cv.value }` (i.e. the more recent key overrides the previous one, if any ... just like the `Map`).
Thoughts?

Reply:
Total code:
```javascript
publication.Metadata.Title = nameObjArray.reduce((pv, cv: IW3cName) => {
if (!cv.value || !cv.language) {
    return pv;
}
return { ...pv, [cv.language]: cv.value };
}, {});
```

Reply:
What is your analysis about `Object.fromEntries()` (which in my view, is quite a heavyweight solution) vs. `array.reduce()` (which I think is a more lightweight and elegant solution, yielding more readable code)

It is absolutely fine if you have a different opinion / coding preference :)
I just want to understand the merit of the proposed technical solution.

Reply:
yes you're right. Your solution is much better - simpler
Thanks you

the running time should be quite similar.

Reply:
In principle, each `array.reduce()` iteration allocates a new object in the memory stack (or is it the heap / somewhere in the closure ... I'm not sure), but dealing with a `Map` and `Object.fromEntries()` to finalize the object from an array structure is costly too .. I guess.